src/ring_theory/polynomial/chebyshev/defs.lean

```diff
@@ -112,4 +126,167 @@ begin
 end
 
 
+end polynomial
+
+namespace polynomial
+
+variables (R S : Type*) [comm_ring R] [comm_ring S]
+
+/-- `chebyshev₂ n` is the `n`-th Chebyshev polynomial of the second kind -/
+noncomputable def chebyshev₂ : ℕ → polynomial R
+| 0       := 1
+| 1       := 2 * X
+| (n + 2) := 2 * X * chebyshev₂ (n + 1) - chebyshev₂ n
+
+@[simp] lemma chebyshev₂_zero : chebyshev₂ R 0 = 1 := rfl
+@[simp] lemma chebyshev₂_one : chebyshev₂ R 1 = 2 * X := rfl
+lemma chebyshev₂_two : chebyshev₂ R 2 = 4 * X ^ 2 - 1 :=
+begin
+  simp only [chebyshev₂],
+  ring,
+end
+
+@[simp] lemma chebyshev₂_add_two (n : ℕ) :
+  chebyshev₂ R (n + 2) = 2 * X * chebyshev₂ R (n + 1) - chebyshev₂ R n :=
+by rw chebyshev₂
+
+lemma chebyshev₂_of_two_le (n : ℕ) (h : 2 ≤ n) :
+  chebyshev₂ R n = 2 * X * chebyshev₂ R (n - 1) - chebyshev₂ R (n - 2) :=
+begin
+  obtain ⟨n, rfl⟩ := nat.exists_eq_add_of_le h,
+  rw add_comm,
+  exact chebyshev₂_add_two R n
+end
+
+lemma chebyshev₂_eq_X_mul_chebyshev₂_add_chebyshev₁ :
+∀ (n : ℕ), chebyshev₂ R (n+1) = X * chebyshev₂ R n + chebyshev₁ R (n+1)
+|0        := by {simp only [chebyshev₂_zero, chebyshev₂_one, chebyshev₁_one], ring}
+|1        := by {simp only [chebyshev₂_one, chebyshev₁_two, chebyshev₂_two], ring}
+|(n + 2)  := begin
+  calc chebyshev₂ R (n + 2 + 1) = 2 * X * (X * chebyshev₂ R (n + 1) + chebyshev₁ R (n + 2))
+                                          - (X * chebyshev₂ R n + chebyshev₁ R (n + 1))
+                : by simp only [chebyshev₂_add_two, chebyshev₂_eq_X_mul_chebyshev₂_add_chebyshev₁ n,
+                                chebyshev₂_eq_X_mul_chebyshev₂_add_chebyshev₁ (n + 1)]
+  ... = X * (2 * X * chebyshev₂ R (n + 1) - chebyshev₂ R n)
+          + (2 * X * chebyshev₁ R (n + 2) - chebyshev₁ R (n + 1))
+                : by ring
+  ... = X * chebyshev₂ R (n + 2) + chebyshev₁ R (n + 2 + 1)
+                : by simp only [chebyshev₂_add_two, chebyshev₁_add_two]
+end
+
+lemma chebyshev₁_eq_chebyshev₂_sub_X_mul_chebyshev₂ (n : ℕ) :
+chebyshev₁ R (n+1) = chebyshev₂ R (n+1) - X * chebyshev₂ R n :=
+by rw [chebyshev₂_eq_X_mul_chebyshev₂_add_chebyshev₁, add_comm (X * chebyshev₂ R n), add_sub_cancel]
+
+
+lemma chebyshev₁_eq_X_mul_chebyshev₁_sub_pol_chebyshev₂ :
+∀ (n : ℕ), chebyshev₁ R (n+2) = X * chebyshev₁ R (n+1) - (1 - X ^ 2) * chebyshev₂ R n
+|0        := by {simp only [chebyshev₁_one, chebyshev₁_two, chebyshev₂_zero], ring}
+|1        := by {simp only [chebyshev₁_add_two, chebyshev₁_zero, chebyshev₁_add_two,
+                              chebyshev₂_one, chebyshev₁_one], ring }
+|(n + 2)  := begin
+calc chebyshev₁ R (n + 2 + 2) = 2 * X * chebyshev₁ R (n + 2 + 1) - chebyshev₁ R (n + 2)
+                                : chebyshev₁_add_two _ _
```

Comment:
```suggestion
calc chebyshev₁ R (n + 2 + 2)
    = 2 * X * chebyshev₁ R (n + 2 + 1) - chebyshev₁ R (n + 2) : chebyshev₁_add_two _ _
```
I think I would prefer less indentation in the rest of the calc block. (Sometimes it's just hard to make it look nice. Don't spend too much time on it.)